### PR TITLE
Hangman: Support tagging premade phrases

### PR DESCRIPTION
**Pull request by mia-pi-git**
https://www.smogon.com/forums/threads/hangman-groups-tags.3680445/#post-8793303

### CODE REVIEW

File: server/chat-plugins/hangman.ts [hunk 1]
```diff
@@ -8,10 +8,30 @@ const HANGMAN_FILE = 'config/chat-plugins/hangman.json';
...
+	tags?: string[];
+}
+
+export let hangmanData: {[roomid: string]: {[phrase: string]: HangmanEntry}} = {};
 
 try {
 	hangmanData = JSON.parse(FS(HANGMAN_FILE).readSync());
+	const roomids = Object.keys(hangmanData);
+	let save = false;
+	for (const roomid of roomids) {
```
Comment (Zarel) [inline server/chat-plugins/hangman.ts:22]:
```suggestion
	let save = false;
	for (const roomid in hangmanData) {
```

**Comment by Zarel** [comment]
Should tags be IDs? That might protect against certain mistakes.

**Comment by Zarel** [comment]
You're certainly using `toID` to _compare_ them. It seems to me they should be stored as IDs so people know what's going on.

**Comment by mia-pi-git** [comment]
> You're certainly using `toID` to _compare_ them. It seems to me they should be stored as IDs so people know what's going on.

I was thinking it might be useful to have them not be IDs, for visual purposes, but this is also a good point. I'll convert it in a moment.

**Comment by mia-pi-git** [comment]
Done.

**Comment by Zarel** [comment]
> I was thinking it might be useful to have them not be IDs, for visual purposes, but this is also a good point. I'll convert it in a moment.

Yeah, I guessed it might be something like that. But for instance Twitter hashtags are ID-like. I think it's probably good for tags to be IDs so people have a visual representation of how they work.

File: server/chat-plugins/hangman.ts (hunk 1 above)
Comment (Zarel) [reply server/chat-plugins/hangman.ts:22]: .

**Pull request merged**